app/models/qa/linked_data/config/context_property_map.rb

```diff
@@ -35,30 +35,55 @@ def initialize(property_map = {}, prefixes = {})
           @ldpath = Qa::LinkedData::Config::Helper.fetch_required(property_map, :ldpath, false)
           @selectable = Qa::LinkedData::Config::Helper.fetch_boolean(property_map, :selectable, false)
           @drillable = Qa::LinkedData::Config::Helper.fetch_boolean(property_map, :drillable, false)
+          @expansion_label_ldpath = Qa::LinkedData::Config::Helper.fetch(property_map, :expansion_label_ldpath, nil)
+          @expansion_id_ldpath = Qa::LinkedData::Config::Helper.fetch(property_map, :expansion_id_ldpath, nil)
           @prefixes = prefixes
         end
 
         # Can this property be the selected value?
-        # @return true if can be selected; otherwise, false
+        # @return [Boolean] true if this property's value can be selected; otherwise, false
         def selectable?
           @selectable
         end
 
         # Can this property be used as a new query
-        # @return true if can be selected; otherwise, false
+        # @return [Boolean] true if this property's value can be used to drill up/down to another level; otherwise, false
         def drillable?
           @drillable
         end
 
+        def group?
+          group_id.present?
+        end
+
+        # Should this URI value be expanded to include its label?
+        # @return [Boolean] true if this property's value is expected to be a URI and its label should be included in the value; otherwise, false
+        def expand_uri?
+          expansion_label_ldpath.present?
+        end
+
+        # Values of this property for a specfic subject URI
+        # @return [Array<String>] values for this property
         def values(graph, subject_uri)
-          output = ldpath_program.evaluate subject_uri, graph
-          output.present? ? output['property'].uniq : nil
-        rescue
-          'PARSE ERROR'
+          ldpath_evaluate(basic_program, graph, subject_uri)
```

Comment:
DRY: Moved the ldpath evaluate process to a private method so it can be shared by multiple methods.